From a2eb46491e28079debc39488a7e23ed76cb5b003 Mon Sep 17 00:00:00 2001
From: Alexandru Isaila <aisaila@bitdefender.com>
Date: Tue, 25 Sep 2018 12:10:38 +0300
Subject: [PATCH] x86/hvm: Add check for cpu_has_vmx_virt_exceptions

This is useful so HVMOP_altp2m_vcpu_enable_notify will fail and not
silently succeed. It save a call to HVMOP_altp2m_set_suppress_ve.

Signed-off-by: Alexandru Isaila <aisaila@bitdefender.com>
Reviewed-by: Andrew Cooper <andrew.cooper3@citrix.com>
diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index 0f910e5..a9449e5 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -4601,6 +4601,12 @@ static int do_altp2m_op(
             break;
         }
 
+        if ( !cpu_has_vmx_virt_exceptions )
+        {
+            rc = -EOPNOTSUPP;
+            break;
+        }
+
         v = d->vcpu[a.u.enable_notify.vcpu_id];
 
         if ( !gfn_eq(vcpu_altp2m(v).veinfo_gfn, INVALID_GFN) ||
